### ISSUE TITLE
is this d-vector embedding input "GE2E Speaker Verification" is moreover same as UIS-RNN or not? 

### ISSUE THREAD

**Issue by MuruganR96** [HarryVolek/PyTorch_Speaker_Verification#5]
sir i have one doubt.

is this d-vector embedding PyTorch implementation of "Generalized End-to-End Loss for Speaker Verification" is moreover common as UIS-RNN or not? 

accuracy for d-vector embeddings?

is it satisfying continuous d-vector embeddings(as sequences) or not? 

sir one you saw this url. i did TIMIT dataset to generated the d-vector embeddings. but how to feed or initialize this embeddings i don't know.

https://github.com/google/uis-rnn/issues/6#issuecomment-439789103

https://github.com/google/uis-rnn/issues/6#issuecomment-439967734

`Some of the libraries are only able to produce per-utterance d-vector embeddings, while for UIS-RNN, we require continuous d-vector embeddings (as sequences). We have no guarantee which third-party library supports this `

can you help me sir? thank you

**Comment by HarryVolek** [comment]
The embeddings produced by this code are currently "utterance" level embeddings, not "segment" level. To go from utterance to segment level d-vectors, the steps under section 2 of https://arxiv.org/abs/1710.10468 will have to be implemented.

Accuracy was 95% when differentiating between 4 speakers at a time.

**Comment by MuruganR96** [comment]
Thank you very much sir.

**Comment by xinli94** [comment]
> The embeddings produced by this code are currently "utterance" level embeddings, not "segment" level. To go from utterance to segment level d-vectors, the steps under section 2 of https://arxiv.org/abs/1710.10468 will have to be implemented.
> 
> Accuracy was 95% when differentiating between 4 speakers at a time.

Hi,

Could you be more specific about the `segment` level embeddings? So do we need to implement this for training, or does it only happen during inference (i.e. while using pretrained model to create the embedding features for an arbitrary length audio input)?

Thanks,
Xin

**Comment by HarryVolek** [comment]
Only during inference.

**Comment by MuruganR96** [comment]
Sir inference means, in UIS-RNN their using d-vector embeddings with
cluster ids for training input.
Sir if we create continues segment and LSTM to get embeddings, apply L2
norms. Then we will easily succeeded UIS-RNN.
Sir any package is there to split up sliding window medhod for the audio
input.?

Thank you so much sir.

**Comment by MuruganR96** [comment]
> Only during inference.

sir @HarryVolek  and @xinli94  

only during inference, we apply clustering to segmenting the utterence, seperating speakers via d-vector embeddings. 
is this correct or not? can you help me sir?

**Comment by MuruganR96** [comment]
sir once again explain segment vs utterance. i have only basic knowledge about these concepts.
how can i change utterance level to segment level embedding?
where i have to change? 
sir can you help me for this concept?
give me some suggestions sir. 
thank you.

**Comment by MuruganR96** [comment]
> Accuracy was 95% when differentiating between 4 speakers at a time.

sir in utterance level is an TD-SV. so it will satisfy for wake-work technique. 
in different speakers wake-word like 'ok google'. will it differentiating speakers at a time?

thank you sir.

**Comment by HarryVolek** [comment]
I added a script to create embeddings compatible with https://github.com/google/uis-rnn.

**Comment by MuruganR96** [comment]
thank you so much sir.
